packages/core/test/acceptance/styling_spec.ts

```diff
@@ -2627,25 +2624,383 @@ describe('styling', () => {
+    @Component({template: '...'})
+    class MyComp {
+      @HostBinding('class') public classes = '';
+    }
+
+    const fixture =
+        TestBed.configureTestingModule({declarations: [MyComp]}).createComponent(MyComp);
+    fixture.detectChanges();
+    const root = fixture.nativeElement as HTMLElement;
+    expect(root.className).toEqual('');
+
+    fixture.componentInstance.classes = '1 2 3';
+    fixture.detectChanges();
+    expect(root.className.split(/\s+/).sort().join(' ')).toEqual('1 2 3');
+
+    root.classList.add('0');
+    expect(root.className.split(/\s+/).sort().join(' ')).toEqual('0 1 2 3');
+
+    // TODO(pk): why ???
+    fixture.componentInstance.classes = '1 2 3 4';
+    fixture.detectChanges();
+    expect(root.className.split(/\s+/).sort().join(' ')).toEqual('0 1 2 3 4');
+  });
+
+  it('should apply camelCased class names', () => {
+    @Component({template: `<div [class]="'fooBar'" [class.barFoo]="true"></div>`})
+    class MyComp {
+    }
+
+    TestBed.configureTestingModule({
+      declarations: [MyComp],
+    });
+    const fixture = TestBed.createComponent(MyComp);
+    fixture.detectChanges();
+
+    const classList = (fixture.nativeElement.querySelector('div') as HTMLDivElement).classList;
+    expect(classList.contains('fooBar')).toBeTruthy();
+    expect(classList.contains('barFoo')).toBeTruthy();
+  });
+
+  it('should convert camelCased style property names to snake-case', () => {
+    @Component({template: `<div [style]="myStyles"></div>`})
+    class MyComp {
+      myStyles = {};
+    }
+
+    TestBed.configureTestingModule({
+      declarations: [MyComp],
+    });
+    const fixture = TestBed.createComponent(MyComp);
+    fixture.detectChanges();
+
+    const div = fixture.nativeElement.querySelector('div') as HTMLDivElement;
+    fixture.componentInstance.myStyles = {fontSize: '200px'};
+    fixture.detectChanges();
+
+    expect(div.style.getPropertyValue('font-size')).toEqual('200px');
+  });
+
+  it('should recover from an error thrown in styling bindings', () => {
+    let raiseWidthError = false;
+
+    @Component({template: `<div [style.width]="myWidth" [style.height]="'200px'"></div>`})
+    class MyComp {
+      get myWidth() {
+        if (raiseWidthError) {
+          throw new Error('...');
+        }
+        return '100px';
+      }
+    }
+
+    TestBed.configureTestingModule({declarations: [MyComp]});
+    const fixture = TestBed.createComponent(MyComp);
+
+    raiseWidthError = true;
+    expect(() => fixture.detectChanges()).toThrow();
+
+    raiseWidthError = false;
+    fixture.detectChanges();
+    const div = fixture.nativeElement.querySelector('div') as HTMLDivElement;
+    expect(div.style.getPropertyValue('width')).toEqual('100px');
+    expect(div.style.getPropertyValue('height')).toEqual('200px');
+  });
+
+  it('should prioritize host bindings for templates first, then directives and finally components',
```

Comment:
This test fails in VE with `Invalid CSS style: unsafe` (but works in ivy) which tells me that there is a difference wrt to sanitisation.

Reply:
This test also fails with ivy with asserts on `elm.style.width`